File: LottieLib_test/app/src/main/java/com/daeseong/lottielib_test/Main5Activity.java
package com.daeseong.lottielib_test;

import androidx.appcompat.app.AppCompatActivity;
import android.animation.Animator;
import android.os.Bundle;
import android.util.Log;
import com.airbnb.lottie.LottieAnimationView;

public class Main5Activity extends AppCompatActivity {

    //애프터이펙트

    private static final String TAG = Main5Activity.class.getSimpleName();

    private LottieAnimationView lottieAnimationView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main5);

        init();

        start();
    }

    @Override
    protected void onStop() {
        super.onStop();

        stop();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        clear();
    }

    private void init(){

        try {

            lottieAnimationView = (LottieAnimationView) findViewById(R.id.lav);
            lottieAnimationView.setAnimation("3.json");
            lottieAnimationView.loop(false);

            Log.e(TAG, "init:" + String.valueOf(lottieAnimationView.getDuration()));

            lottieAnimationView.addAnimatorListener(new Animator.AnimatorListener() {
                @Override
                public void onAnimationStart(Animator animator) {

                }

                @Override
                public void onAnimationEnd(Animator animator) {

                    Log.e(TAG, "onAnimationEnd:" + String.valueOf(lottieAnimationView.getDuration()));
                }

                @Override
                public void onAnimationCancel(Animator animator) {

                }

                @Override
                public void onAnimationRepeat(Animator animator) {

                }
            });

        }catch (Exception ex){
            ex.printStackTrace();
        }
    }

    private void start(){

        try {

            boolean isPlay = lottieAnimationView.isAnimating();
            if (!isPlay) {
                lottieAnimationView.playAnimation();
            }

        }catch (Exception ex){
            ex.printStackTrace();
        }
    }

    private void stop(){

        try {

            boolean isPlay = lottieAnimationView.isAnimating();
            if (isPlay) {
                lottieAnimationView.cancelAnimation();
            }

        }catch (Exception ex){
            ex.printStackTrace();
        }
    }

    private void clear(){

        try {

            if (lottieAnimationView != null) {
                lottieAnimationView.removeAllLottieOnCompositionLoadedListener();
                lottieAnimationView.removeAllAnimatorListeners();
                lottieAnimationView.removeAllLottieOnCompositionLoadedListener();
            }

        }catch (Exception ex){
            ex.printStackTrace();
        }
    }

}

File: lottielib_test_kotlin/app/src/main/java/com/daeseong/lottielib_test/Main5Activity.kt
package com.daeseong.lottielib_test

import android.animation.Animator
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.airbnb.lottie.LottieAnimationView

class Main5Activity : AppCompatActivity() {

    private val tag = Main5Activity::class.java.simpleName

    private lateinit var lottieAnimationView: LottieAnimationView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main5)

        init()

        start()
    }

    override fun onStop() {
        super.onStop()

        stop()
    }

    override fun onDestroy() {
        super.onDestroy()

        clear()
    }

    private fun init() {

        try {
            lottieAnimationView = findViewById(R.id.lav)
            lottieAnimationView.apply {
                setAnimation("3.json")
                loop(false)

                Log.e(this@Main5Activity.tag, "init: $duration")

                addAnimatorListener(object : Animator.AnimatorListener {
                    override fun onAnimationStart(animator: Animator) {}

                    override fun onAnimationEnd(animator: Animator) {
                        Log.e(this@Main5Activity.tag, "onAnimationEnd: $duration")
                    }

                    override fun onAnimationCancel(animator: Animator) {}

                    override fun onAnimationRepeat(animator: Animator) {}
                })
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
    }

    private fun start() {

        try {
            if (!lottieAnimationView.isAnimating) {
                lottieAnimationView.playAnimation()
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
    }

    private fun stop() {

        try {
            if (lottieAnimationView.isAnimating) {
                lottieAnimationView.cancelAnimation()
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
    }

    private fun clear() {

        try {
            lottieAnimationView.apply {
                removeAllLottieOnCompositionLoadedListener()
                removeAllAnimatorListeners()
                removeAllLottieOnCompositionLoadedListener()
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
    }
}